Style enhancements

clarkepaul opened the pull request:
Some adjustments to make things more BEM
Reworked the actions styles a bit but doesn't really conform to BEM yet

chillu commented on admin/client/src/components/Accordion/AccordionGroup.js:
```diff
@@ -15,12 +15,10 @@ class AccordionGroup extends SilverStripeComponent {
-            {this.props.title}
-          </a>
-        </h6>
+      <div className="accordion">
```
This seems wrong, the component is called `AccordionGroup`, but the class is `accordion`. We still have an `Accordion` component as the parent of this one, right?

clarkepaul replied:
Hmm, I think I assumed the component was called Accordion as there is also an Accordion.js file which wraps AccordionGroup.

clarkepaul replied:
Happy to make any amendments you think are needed. It kind of feels like AccordionGroup and Accordion are separate components?

chillu replied:
Caught up with Paul about this offline, we agreed that `AccordionGroup` and `Accordion` belong together, but `AccordionItem` should be a separate component since list item styles should be separate from their placement in an accordion. I think Paul's done some work around renaming `AccordionGroup` to `AccordionBlock` since it is more consistent with bootstrap examples on how to use the collapsible JS behaviour as a de-facto accordion.

clarkepaul commented:
A fair bit of work has been done on the form actions, I've mapped them out as pictured and have some html to paste into for testing purposes.
![pasted_image_27_04_16__9_30_am](https://cloud.githubusercontent.com/assets/555033/14837018/f3a02e04-0c68-11e6-9254-59dee4c6404e.png)

`<div className="btn-tooldar">
  <button type="button" className="btn btn-primary">Primary</button>
  <button type="button" className="btn btn-secondary">Secondary</button>
  <button type="button" className="btn btn-link text-muted">Link muted</button>
  <button type="button" className="btn btn-info">Info</button>
  <button type="button" className="btn btn-warning">Warning</button>
  <button type="button" className="btn btn-danger">Danger</button>
  <button type="button" className="btn btn-link">Link</button>
  <button type="button" className="btn btn-link text-danger">Link danger</button>
  <button type="button" className="btn btn-primary font-icon-rocket">Primary</button>
  <button type="button" className="btn btn-primary font-icon-tick btn--no-text"></button>
</div><br />
<div className="btn-tooldar">
  <button type="button" className="btn btn-primary-outline">Primary-outline</button>
  <button type="button" className="btn btn-secondary-outline">Secondary-outline</button>
  <button type="button" className="btn btn-info-outline">Info-outline</button>
  <button type="button" className="btn btn-warning-outline">Warning-outline</button>
  <button type="button" className="btn btn-danger-outline">Danger-outline</button>
  <button type="button" className="btn btn-primary btn--loading">
    <div className="btn__loading-icon" >
      <svg viewBox="0 0 44 12">
        <circle cx="6" cy="6" r="6" />
        <circle cx="22" cy="6" r="6" />
        <circle cx="38" cy="6" r="6" />
      </svg>
    </div>
  Loading</button>
  <button type="button" className="btn btn-primary-outline btn--loading">
    <div className="btn__loading-icon" >
      <svg viewBox="0 0 44 12">
        <circle cx="6" cy="6" r="6" />
        <circle cx="22" cy="6" r="6" />
        <circle cx="38" cy="6" r="6" />
      </svg>
    </div>
  Loading</button>
  <button type="button" className="btn btn-secondary btn--loading">
    <div className="btn__loading-icon" >
      <svg viewBox="0 0 44 12">
        <circle cx="6" cy="6" r="6" />
        <circle cx="22" cy="6" r="6" />
        <circle cx="38" cy="6" r="6" />
      </svg>
    </div>
  Loading</button>
  <button type="button" className="btn btn-secondary-outline btn--loading">
    <div className="btn__loading-icon" >
      <svg viewBox="0 0 44 12">
        <circle cx="6" cy="6" r="6" />
        <circle cx="22" cy="6" r="6" />
        <circle cx="38" cy="6" r="6" />
      </svg>
    </div>
  Loading</button>
</div>`

clarkepaul commented:
I've done some more refactoring based on discussions above to improve component naming, and a bit more tidy-up of styles to align with BEM. Separated out list-group from the accordion into its own component.

tractorcow commented on admin/client/src/styles/_typography.scss:
```diff
@@ -58,3 +59,10 @@
 	}
 }
+
+.label {
```
Having this at the top level feels very generic.

clarkepaul replied:
Its a bootstrap class which is already top level so if it was being used it would affect things already.

tractorcow commented on admin/client/src/containers/CampaignAdmin/CampaignAdmin.scss:
```diff
@@ -1,103 +1 @@
-.campaign-items {
 
```
May as well delete this file instead of leaving it empty.

tractorcow commented on admin/client/src/components/ListGroup/LIstGroup.scss:
```diff
@@ -0,0 +1,96 @@
+	}
+}
+
+.list-group-item__heading {
```
Bootstrap has a standard `list-group-item-heading` class we should probably use.

tractorcow commented on admin/client/src/components/ListGroup/LIstGroup.scss:
```diff
@@ -0,0 +1,96 @@
+// List-group, based on Bootstraps list-group
+// TODO split out campaign specific css from generic list-group
+
+.list-group {
```
This file has the first two letters capitalised by mistake. you can use `git mv LIstGroup.scss ListGroup.scss` to fix.

clarkepaul replied:
fatal: Bad source. Should I just rename manually?

clarkepaul replied:
I made the other amendments but the file name seems like it hasn't updated, I take it this is something you can do @tractorcow ?

tractorcow commented on admin/client/src/components/ListGroup/LIstGroup.scss:
```diff
@@ -0,0 +1,96 @@
+	}
+}
+
+.list-group-item--published {
```
Anything about `published` etc is specific to campaigns. Should this belong in the generic ListGroup component? Maybe put this back into the campaign admin area?

clarkepaul replied:
Yup true but then it wouldn't be attached to the list-group-item? because published is a state which can also be applied outside of campaigns. I felt it was something I didn't completely know how to resolve in a nice way yet so made a comment at the top "// TODO split out campaign specific css from generic list-group"

clarkepaul commented:
I'm in the process of separating the list-group styles and campaign styles apart. Don't merge yet

clarkepaul commented:
Ok I've done some more refactoring which splits out list-group into its own component but there is possibly some more work React wise to split it out of campaign-admin completely. I think I've addressed all the issues mentioned above and I've hidden the campaign-item links which aren't needed for Alpha1.
